db/db_impl_compaction_flush.cc

```diff
@@ -1115,6 +1073,59 @@ Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
   return s;
 }
 
+Status DBImpl::WaitUntilFlushWouldNotStallWrites(ColumnFamilyData* cfd) {
+  {
+    InstrumentedMutexLock l(&mutex_);
+    uint64_t orig_active_memtable_id = cfd->mem()->GetID();
+    WriteStallCondition write_stall_condition = WriteStallCondition::kNormal;
+    do {
+      if (write_stall_condition != WriteStallCondition::kNormal) {
+        TEST_SYNC_POINT("DBImpl::CompactRange:StallWait");
```

Comment:
Note that the name of the TEST_SYNC_POINT should be adjusted before merging this PR.